### DISCUSSION TITLE
useNavigation hook does not include replace

### DISCUSSION THREAD

**Discussion by react-ghost** [react-navigation/react-navigation#10758] [Q&A]
if I instantiate a navigation object like below:

`const navigation = useNavigation()`

The returned object does not have the 'replace' function that the objects provided to child components have when using them inside any type of navigator (stack, tab, etc.)

Is there anyway I can get this function from this hook, and why is this the case?

**Comment by satya164** [comment]
`useNavigation` provides the `navigation` object for the current screen. It cannot give you the `navigation` object from a child screen, neither can the `navigation` prop.

**Reply by cetfu** [reply]
should we use StackActions?

**Reply by satya164** [reply]
it's not supported unless you explicitly provide the `key` of the navigation state belonging to the navigator in the `target` property https://reactnavigation.org/docs/navigation-object/#dispatch

and if you do that, then you should have proper tests to verify the functionality. react navigation's API doesn't guarantee that a navigator will handle the action in every scenario, nor can it be type-checked - which makes refactors risky without proper tests.
